### ISSUE TITLE
New defaults for the config

### ISSUE THREAD

**Issue by szotsaki** [Kozea/Radicale#346]
I've packaging Radicale for about 4 years now for openSUSE and I always patch the default config for a more secure one.

Could you please take a look at its diff and transfer the changes you also agree? Thank you!

``` diff
--- config  2015-01-21 09:42:40.966800256 +0100
+++ config  2015-01-21 09:48:16.530176100 +0100
@@ -16,25 +16,25 @@
 # IPv6 syntax: [address]:port
 # For example: 0.0.0.0:9999, [::]:9999
 # IPv6 adresses are configured to only allow IPv6 connections
-#hosts = 0.0.0.0:5232
+hosts = 127.0.0.1:5232

 # Daemon flag
-#daemon = False
+daemon = True

 # File storing the PID in daemon mode
 #pid =

 # SSL flag, enable HTTPS protocol
-#ssl = False
+#ssl = True

 # SSL certificate path
-#certificate = /etc/apache2/ssl/server.crt
+#certificate = /etc/radicale/radicale.cert.pem

 # SSL private key
-#key = /etc/apache2/ssl/server.key
+#key = /etc/radicale/radicale.key.pem

 # SSL Protocol used. See python's ssl module for available values
-#protocol = PROTOCOL_SSLv23
+#protocol = PROTOCOL_TLSv1_2

 # Ciphers available. See python's ssl module for available ciphers
 #ciphers =
@@ -74,7 +74,7 @@

 # Authentication method
 # Value: None | htpasswd | IMAP | LDAP | PAM | courier | http | remote_user | custom
-#type = None
+type = htpasswd

 # Custom authentication handler
 #custom_handler =
@@ -84,7 +84,7 @@

 # Htpasswd encryption method
 # Value: plain | sha1 | ssha | crypt | bcrypt | md5
-#htpasswd_encryption = crypt
+htpasswd_encryption = bcrypt

 # LDAP server URL, with protocol and port
 #ldap_url = ldap://localhost:389/
@@ -140,13 +140,13 @@

 # Rights backend
 # Value: None | authenticated | owner_only | owner_write | from_file | custom
-#type = None
+type = owner_only

 # Custom rights handler
 #custom_handler =

 # File for rights management from_file
-#file = ~/.config/radicale/rights
+file = /etc/radicale/rights


 [storage]
@@ -163,7 +163,7 @@
 #custom_handler =

 # Folder for storing local collections, created if not present
-#filesystem_folder = ~/.config/radicale/collections
+filesystem_folder = /var/lib/radicale/calendars

 # Database URL for SQLAlchemy
 # dialect+driver://user:password@host/dbname[?key=value..]
```

(Some lines change default paths, I don't know whether you want to apply them.)

**Comment by movatica** [comment]
I partially agree with the changes, though not all of them:

- Setting the host to the localhost ip restricts remote clients from connecting, which is only useful if the server is used behind a frontend server like nginx. I'd recommend keeping the 0.0.0.0 value.
- What is the use of adding .pem file suffixes for the SSL keys instead of keeping the default endings? Though I agree with the different path.
- The suggested filesystem folder is against the FHS as described in http://www.pathname.com/fhs/pub/fhs-2.3.html. I'd rather use ~/collections, as the daemon should have it's own user anyway.

I'd also recommend to provide a default for the ciphers setting, as this is really difficult to get right from all the different documentation and choices... After scrolling through the Python and OpenSSL docs, I ended up with ciphers = TLSv1_2:!eNULL, but still have to check whether it works the way I want it to.

For the pid setting, the straight forward value is /var/run/radicale.pid

**Comment by szotsaki** [comment]
- I also agree with keeping the 0.0.0.0 value.
- .pem ending denotes the encoding type, that's why I thought of it. Let's Encrypt also has [this kind of notation](https://letsencrypt.readthedocs.org/en/latest/using.html#where-are-my-certificates).
- The default cipher list is a tough choice. HTTP/2 has a [list of bad cipher suites](https://http2.github.io/http2-spec/#BadCipherSuites) which "[...] do not offer an ephemeral key exchange and those that are based on the TLS null, stream, or block cipher type".
- I also would like to revert the "daemon=True" change because systemd requires the program to run in the foreground. I recently migrated Radicale on openSUSE to systemd and I also made it explicit to run in foreground.
